source/modules/water/memory/ByteOrder.h

```diff
@@ -153,7 +153,7 @@ inline uint32 ByteOrder::swap (uint32 n) noexcept
 {
    #ifdef CARLA_OS_MAC
     return OSSwapInt32 (n);
-   #elif defined(CARLA_OS_WIN) || ! (defined (__arm__) || defined (__arm64__))
+   #elif defined(CARLA_OS_WIN) || ! (defined (__arm__) || ! defined (__arm64__))

```

Comment:
hmm this logic seems completely wrong...
how does this even work.

before we had, in pseudo-code:
if WINDOWS or not (arm or arm64)

now changed to:
if WINDOWS or not (arm or not arm64)

seems completely to go against the title you put on this PR.

Reply:
You are right. Now I'm confused why it errored at the next line when compiling for arm64 and now it does compile fine.